File: solutions/elixir/rna-transcription/1/lib/rna_transcription.ex
defmodule RnaTranscription do
  @dna_to_rna %{
    ?A => ?U,
    ?C => ?G,
    ?G => ?C,
    ?T => ?A,
  }

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

    iex> RnaTranscription.to_rna(~c"ACTG")
    ~c"UGAC"
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna), do: Enum.map(dna, & @dna_to_rna[&1])
end

File: solutions/typescript/rna-transcription/1/rna-transcription.ts
const dnaToRna: { [key: string]: string } = {
  'A': 'U',
  'C': 'G',
  'G': 'C',
  'T': 'A',
}

export function toRna(dna: string): string {
  return [...dna].map(c => {
    if (c in dnaToRna) return dnaToRna[c];
    throw new Error(`Invalid input DNA.`);
  }).join('');
}
